### PR TITLE
Add some package and resource checks

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
- Refine workspace connection provider error handling
- Fix run name get dir name empty error
- Limit workspace type to 'project' and 'default' if use workspace connection

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/operations/_local_azure_connection_operations.py [hunk 1]
```diff
@@ -15,13 +15,21 @@
...
 )
+from promptflow.core._errors import MissingRequiredPackage
 from promptflow.core._utils import extract_workspace
 
 logger = get_cli_sdk_logger()
 
 
 class LocalAzureConnectionOperations(WorkspaceTelemetryMixin):
     def __init__(self, connection_provider, **kwargs):
+        try:
```
Comment (crazygao) [inline src/promptflow-devkit/promptflow/_sdk/operations/_local_azure_connection_operations.py:26]: leave this to me. I make it in the test movement.

File: src/promptflow-core/promptflow/_constants.py [hunk 2]
```diff
@@ -277,6 +277,7 @@ class ConnectionProviderConfig:
     "(/providers/Microsoft.MachineLearningServices)?/workspaces/([^/]+)$"
 )
 CONNECTION_DATA_CLASS_KEY = "DATA_CLASS"
+AML_CONNECTION_PROVIDER_TEMPLATE = "azureml://subscriptions/{}/resourceGroups/{}/providers/Microsoft.MachineLearningServices/workspaces/{}"  # noqa: E501
```
Comment (0mza987) [inline src/promptflow-core/promptflow/_constants.py:280]: Change to `AML_PROVIDER_TEMPLATE`?

File: src/promptflow-core/promptflow/core/_connection_provider/_utils.py [hunk 3]
```diff
@@ -67,3 +67,14 @@ def interactive_credential_disabled():
 def is_from_cli():
     """Check if the current execution is from promptflow-cli."""
     return "promptflow-cli" in ClientUserAgentUtil.get_user_agent()
+
+
+def check_connection_provider_resource(resource_id: str, credential, pkg_name):
+    from .._utils import get_workspace_from_resource_id
+
+    workspace = get_workspace_from_resource_id(resource_id, credential, pkg_name)
+    if workspace._kind not in [AzureWorkspaceKind.DEFAULT, AzureWorkspaceKind.PROJECT]:
```
Comment (0mza987) [inline src/promptflow-core/promptflow/core/_connection_provider/_utils.py:76]: extract this to a util func?

File: src/promptflow-core/promptflow/_constants.py (hunk 2 above)
Comment (brynn-code) [reply src/promptflow-core/promptflow/_constants.py]: Changed to AML_WORKSPACE_TEMPLATE

File: src/promptflow-devkit/promptflow/_sdk/entities/_run.py [hunk 4]
```diff
@@ -515,7 +515,7 @@ def _format_display_name(self) -> str:
 
     def _get_flow_dir(self) -> Path:
         if not self._use_remote_flow:
-            flow = Path(self.flow)
+            flow = Path(str(self.flow)).resolve().absolute()
```
[the remark this answers is not in the document]
Comment (brynn-code) [reply src/promptflow-devkit/promptflow/_sdk/entities/_run.py:518]: "." have empty name.

File: src/promptflow-core/promptflow/core/_connection_provider/_utils.py (hunk 3 above)
Comment (brynn-code) [reply src/promptflow-core/promptflow/core/_connection_provider/_utils.py:76]: offline synced.

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow-devkit/promptflow/_sdk/entities/_run.py (hunk 4 above)
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/entities/_run.py:518]: may add a changelog for this fix : )

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,10 @@
 ## 1.9.0 (Upcoming)
 
 ### Features Added
-- [CLI]: Added autocomplete feature.
+- [promptflow-devkit]: Added autocomplete feature, reach [here](https://microsoft.github.io/promptflow/reference/pf-command-reference.html#autocomplete) for more details.
```
Comment (wangchao1230) [inline src/promptflow/CHANGELOG.md:6]: Added autocomplete feature for linux

**Review by crazygao — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**